src/services/taskService.js

```diff
@@ -0,0 +1,302 @@
+//importing files...
+import api from "./axiosConfig";
+import { v4 as uuidv4 } from "uuid";
+
+
+//GET ENDPOINTS
+/**
+ * getAllTasks - Asynchronous function that gets all tasks in the db
+ * @param {object} queryParams: An object that contains the query parameters to retrieve tasks with
+ *                              or defaults to an empty object
+ * @returns {Promise<Array>} All tasks created
+ */
+export const getAllTasks = async (queryParams = {}) => {
+  try {
+    //Building the queryString...
+    const queryString = new URLSearchParams(queryParams).toString();
+    const url = queryString ? `/tasks?${queryString}` : "/tasks";
+    const res = await api.get(url);
+    return res.data;
+  } catch (err) {
+    console.error("Error retrieving tasks:", err);
+    throw err;
+  }
+};
+
+/**
+ * getTask - Asynchronous function that gets a single task by ID
+ * @param {String} taskId: ID of task to retrieve
+ * @returns {object} The task object received or throws error
+ */
+export const getTask = async (taskId) => {
+  try {
+    const res = await api.get(`/tasks/${taskId}`);
+    //check status of response
+    if (res.status === 500) {
+      throw new Error("Server error");
+    }
+    //retrieve task from response
+    const task = res.data;
+    if (!task) {
+      throw new Error("Task not found");
+    }
+    return task;
+  } catch (err) {
+    console.error("Error retrieving task: ", err);
+    throw err;
+  }
+};
+
+/**
+ * getAllSubtasks - Async function that retrieves all the subtasks for a task
+ * @param {String} taskId: ID of tasks for which subtasks is to be retrieved
+ * @returns {Promise<Array>} An array of subtasks
+ */
+export const getAllSubTasks = async (taskId) => {
+  try {
+    const task = await getTask(taskId);
+    return task.subtasks;
+  } catch (err) {
+    console.error("Error getting subtasks: ", err);
+    throw err;
+  }
+};
+
+/**
+ * getSubtask - Async function that retrieves a single subtask
+ * @param {String} taskId ID of task containing the subtask to be retrieved
+ * @param {String} subtaskId ID of subtask to be retrieved from subtasks array
+ * @returns {object} subtask to be retrieved
+ */
+export const getSubtask = async (taskId, subtaskId) => {
```

Comment:
Is there any situation where we'd need to get a specific subtask?

The Figma design has a section that shows all subtasks at once. Same applies with notes. Unless we have a specific subtask detail page; this might not be needed.

Do you understand?

Reply:
Currently, there's no situation that will need it. I will take out the methods.